internal/seclang/rules_test.go

```diff
@@ -885,27 +897,49 @@ func TestParameterPollution(t *testing.T) {
 		t.Errorf("failed to test arguments pollution: Single match fixed case: %d, %+v\n",
 			len(tx.MatchedRules()), tx.MatchedRules())
 	}
+}
 
-	tx = waf.NewTransaction()
+// HPP - Detect HTTP Parameter Pollution Attacks
+// This test case uses two rules instead of one to test the parameter pollution. The rules are:
+// 1. SecRule ARGS:test1 "xyz" "id:3, phase:2, log, pass"
+// 2. SecRule ARGS:test1 "ZZZZ" "id:4, phase:2, log, pass"
+// Attack:
+// POST /test?test1=xyz&test1=ABCD
+// test1=abc&test1=ZZZZ
+// In this case, the attacker tries to send two different values for the same parameter, and the rule should match in
+// both cases. Coraza should add the matched parameter to an array an iterate over it to check for matches.x`
+func TestMultipleParameterPollution(t *testing.T) {
+	rules := `SecRule ARGS:test1 "xyz" "id:3, phase:2, log, pass"
+SecRule ARGS:test1 "ZZZZ" "id:4, phase:2, log, pass"`
+	waf := corazawaf.NewWAF()
+	parser := NewParser(waf)
+	err := parser.FromString(rules)
+	if err != nil {
+		t.Error()
+		return
+	}
+	tx := waf.NewTransaction()
 	tx.ProcessRequestHeaders()
-	tx.AddPostRequestArgument("test1", "xyz")
-	tx.AddPostRequestArgument("Test1", "Xyz")
-	tx.AddPostRequestArgument("tesT1", "Xyz")
-	tx.AddPostRequestArgument("TEST1", "XYZ")
+	tx.AddGetRequestArgument("test1", "xyz")
+	tx.AddGetRequestArgument("test1", "ABCD")
+	tx.AddPostRequestArgument("test1", "abc")
+	tx.AddPostRequestArgument("test1", "ZZZZ")
```

Comment:
```suggestion
	tx.AddGetRequestArgument("test1", "xyz")
	tx.AddGetRequestArgument("test1", "ABCD")
	tx.ProcessRequestHeaders()
	tx.AddPostRequestArgument("test1", "abc")
	tx.AddPostRequestArgument("test1", "ZZZZ")
```
I think that the multiphase test is failing because request args are checked asap (during phase1). The test adds the get args after calling processrequestheaders, therefore when phase1 is checked no args_get are present and the xyz rule does not match anything